service/storage_service.cc

```diff
@@ -4062,6 +4062,11 @@ future<> storage_service::raft_removenode(locator::host_id host_id, std::list<lo
         group0_command g0_cmd = _group0->client().prepare_command(std::move(change), guard, ::format("removenode: request remove for {}", id));
 
         request_id = guard.new_group0_state_id();
+
+        if (auto itr = _topology_state_machine._topology.requests.find(id);
+                itr != _topology_state_machine._topology.requests.end() && itr->second == topology_request::remove) {
+            throw std::runtime_error("Removenode failed. Concurrent request for removal already in progress");
```

Comment:
Yeah, I reproduced it, the test hanged before this change. So the issue was not in the toplogy coordinator, it was that during the enqueue operation, the second topology change request append was causing error(line 4075). So throwing the error before that resolves the issue.

Reply:
What I meant is whether you managed to reproduce the scenario described by @patjed41 in issue https://github.com/scylladb/scylladb/issues/20271, i.e.: we try to remove a normal node, and it is normal when we initiate request, but it is no longer normal when topology coordinator actually handles the request.

It looks like you reproduced some other scenario.

So my question is whether the exact scenario described by @patjed41 is impossible or maybe it just requires different timing to hit it?

Reply:
I would say its impossible, because in that case, the second removal request overwrites the first one, and both of them waits for the request to complete and only one succeeds, the second one waits forever. cc - @gleb-cloudius 

Regarding timing, there are 2 cases:

Case 1: 2nd node executes raft_removenode when node1's request is in queue.
This is above scenario.

Case 2: 2nd node executes raft_removenode when node1's request has succeeded.
In this case the error is thrown in raft_removenode because node3 is no longer normal.

Reply:
> I would say its impossible

Node been not normal mat be impossible, but there are some other checks here such as node is the last one with tokens for instance. I think this is possible to hit.